users/views.py

```diff
@@ -106,54 +107,50 @@ def get(self: "ActivateUser", request: HttpRequest, uidb64: str, token: str) ->
-        return Profile.objects.get(user=self.request.user)
-
-    def get(self: "ProfileSettingsView") -> HttpResponse:
-        self.object = self.get_object()
-        user = self.object.user
-
-        user_form = UserForm(instance=user)
-        profile_form = self.form_class(instance=self.object)
-
-        return self.render_to_response(self.get_context_data(user_form=user_form, profile_form=profile_form))
-
-    def post(self: "ProfileSettingsView", request: HttpRequest) -> HttpResponse:
-        self.object = self.get_object()
-        user = self.object.user
-
-        user_form = UserForm(request.POST, instance=user)
-        profile_form = self.form_class(request.POST, request.FILES, instance=self.object)
+    def get_form(self: "ProfileSettingsView") -> dict[str, Any]:
+        return {
+            "user_form": UserForm(instance=self.request.user),
+            "profile_form": self.form_class(instance=self.request.user.profile),
+        }
 
+    def form_valid(self: "ProfileSettingsView", form: dict[str, Any]) -> HttpResponse:
+        user_form = form["user_form"]
+        profile_form = form["profile_form"]
         if user_form.is_valid() and profile_form.is_valid():
             user_form.save()
             profile_form.save()
-            return redirect(self.success_url)
-
-        return self.render_to_response(self.get_context_data(user_form=user_form, profile_form=profile_form))
+            return super().form_valid(form)
+        return self.form_invalid(form)
 
-    def get_context_data(self: "ProfileSettingsView", **kwargs: dict[str, Any]) -> dict:
-        context = super().get_context_data(**kwargs)
-        if "user_form" not in context:
-            context["user_form"] = UserForm(instance=self.request.user)
-        if "profile_form" not in context:
-            context["profile_form"] = self.form_class(instance=self.get_object())
-        return context
+    def post(self: "ProfileSettingsView", request: HttpRequest, *args: tuple, **kwargs: dict[str, Any]) -> HttpResponse:  # noqa: ARG002
+        with transaction.Atomic():
+            form = self.get_form()
+            form["user_form"] = UserForm(request.POST, instance=request.user)
+            form["profile_form"] = self.form_class(request.POST, request.FILES, instance=request.user.profile)
+            if self.form_valid(form):
+                return self.form_valid(form)
+            return self.form_invalid(form)
 
 
-class AccountSettingsView(LoginRequiredMixin, UpdateView):
+class AccountSettingsView(LoginRequiredMixin, FormView):
     model = UserSettings
     template_name = "users/account_settings.html"
     form_class = UserSettingsForm
     success_url = reverse_lazy("profile")
     login_url = "login"
 
-    def get_object(self: "AccountSettingsView") -> UserSettings:
-        return UserSettings.objects.get(user=self.request.user)
+    def form_valid(self: "AccountSettingsView", form: UserSettingsForm) -> HttpResponse:
```

Comment:
I think we don't need this
https://github.com/django/django/blob/3fad712a91a8a8f6f6f904aff3d895e3b06b24c7/django/views/generic/edit.py#L131
Django does the same.

Reply:
It is formView which inherit from FormMixin instead of ModelFormMixin :( and in FormMixin is form_valid without save 
![image](https://github.com/user-attachments/assets/d87be390-34d9-47b5-a145-54b688afffcd)